brave/brave-core
# Fix #11444: Google accounts should not show in the profile menu

Resolves https://github.com/brave/brave-browser/issues/11444

## Description

1. This PR moves the toggle to enable `Google Login for extensions` to `brave://settings/extensions` which is disabled by default.
2. `ProfileViewMenu` should not display google profile info when the user has enabled `Google Login` for extensions. 

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

1. Launch browser with a clean profile. Login to `gmail.com`.
2. Navigate to brave://signin-internals and verify there are no `Accounts in Token Service` or `Refresh token events`
3. Navigate to brave://settings/extensions - Toggle `Google Login for extensions` and Relaunch
4. Navigate to https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi - Install and verify login works
5. Navigate to https://chrome.google.com/webstore/detail/google-calendar/gmbgaklkmjakoegficnlkhebmhkjfich - Install and verify that login works
6. Add a new profile `Profiles > Add Profile` and verify that the `Profile Menu` next to the hamburger menu doesn't show your google login information. 
7. Navigate to brave://settings/siteData and delete all site data
8. Navigate to brave://settings/extensions - Toggle `Google Login for extensions` and Relaunch
9. Navigate to https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi - Install and `Sign In` will not work.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
